x/leverage/keeper/inspector.go

```diff
@@ -0,0 +1,363 @@
+func (q Querier) InspectNeat(
+	goCtx context.Context,
+	req *types.QueryInspectNeat,
+) (*types.QueryInspectNeatResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	// The "all" symbol denom is converted to empty symbol denom
+	if strings.EqualFold(req.Symbol, "all") {
+		req.Symbol = ""
+	}
+
+	req2 := types.QueryInspect{
+		Symbol:  req.Symbol,
+		Mode:    req.Mode,
+		Sort:    req.Sort,
+		ModeMin: req.ModeMin,
+		SortMin: req.SortMin,
+	}
+
+	resp2, err := q.Inspect(goCtx, &req2)
+	if err != nil {
+		return nil, err
+	}
+
+	borrowers := []types.BorrowerSummaryNeat{}
+	for _, b := range resp2.Borrowers {
+		if b.SuppliedValue.IsPositive() {
+			borrowed := b.BorrowedValue
+			if req.Symbol != "" {
+				borrowed = b.SpecificBorrowValue
+			}
+			neat := types.BorrowerSummaryNeat{
+				Account:  b.Address,
+				Borrowed: neat(borrowed),
+				LTB:      neat(b.BorrowedValue.Quo(b.BorrowLimit)),
+				LTL:      neat(b.BorrowedValue.Quo(b.LiquidationThreshold)),
+				LTV:      neat(b.BorrowedValue.Quo(b.CollateralValue)),
+			}
+			borrowers = append(borrowers, neat)
+		}
+	}
+
+	return &types.QueryInspectNeatResponse{Borrowers: borrowers}, nil
+}
+
+// Separated from grpc_query.go
+func (q Querier) RiskData(
+	goCtx context.Context,
+	req *types.QueryRiskData,
+) (*types.QueryRiskDataResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	// This query is also disabled by default as a safety measure. Enable with liquidator queries.
+	if !q.Keeper.liquidatorQueryEnabled {
+		return nil, types.ErrNotLiquidatorNode
+	}
+	k, ctx := q.Keeper, sdk.UnwrapSDKContext(goCtx)
+	borrowers := k.riskData(ctx)
+
+	return &types.QueryRiskDataResponse{Borrowers: borrowers}, nil
+}
+
+// neat truncates an sdk.Dec to a common-sense precision based on its size and converts it to float.
+// This makes a big difference in readability when using the CLI.
+func neat(num sdk.Dec) float64 {
+	n := num.MustFloat64()
+	precision := 3 // Round to thousandths if 0.001 <= n <= 10
+	if n > 10 {
+		precision = 1 // above $10: Round to dime
+	}
+	if n > 1000 {
+		precision = 0 // above $1000: Round to dollar
+	}
+	if n > 1_000_000 {
+		precision = -3 // above $1000000: Round to thousand
+	}
+	if n < 0.001 {
+		precision = 6 // round to millionths
+	}
+	if n < 0.000001 {
+		return n // tiny: maximum precision
+	}
+	// Truncate the float at a certain precision (can be negative)
+	x := math.Pow(10, float64(precision))
+	return float64(int(n*x)) / x
+}
+
+// filteredSortedBorrowers returns a list of borrower addresses and their account summaries sorted and filtered
+// by selected methods. Sorting is ascending in X if the sort function provided is "less X", and descending in
+// X if the function provided is "more X"
+func (k Keeper) filteredSortedBorrowers(
+	ctx sdk.Context, filters []inspectorFilter, sorting inspectorSort, symbol string,
```

Comment:
let's document the `symbol`.

Reply:
now documented in proto (and this sub function with `symbol` param no longer exists)